Sources/DeploymentTargetAWSLambda/main.swift

```diff
@@ -0,0 +1,337 @@
+            Also configures an API Gateway to make the Lambda functions accessible over HTTP.
+            """,
+        version: String(LambdaDeploymentProvider.version)
+    )
+    
+    @Argument(help: "Directory containing the Package.swift with the to-be-deployed web service's target")
+    var inputPackageDir: String
+    
+    @Option(help: "Name of the web service's SPM target/product")
+    var productName: String
+    
+    @Option
+    var awsProfileName: String = "default"
+    
+    @Option
+    var awsRegion: String = "eu-central-1"
+    
+    @Option(help: "Name of the S3 bucket to upload the lambda package to")
+    var s3BucketName: String
+    
+    @Option(help: """
+        Path of where in the bucket the lambda package should be stored.
+        If a file already exists at the specified location (e.g. from a previous deployment) it will be overwritten.
+        Note that this is only the folder path, and should not contain the actual filename.
+        The lambda package will be stored at 's3://{bucket_name}/{bucket_path}/{product_name}.zip', where {product_name} is the name of the web service's target (as specified via the 'product-name' option).
+        """
+    )
+    var s3BucketPath: String = "/apodini-lambda/"
+    
+    @Option
+    var awsApiGatewayApiId: String
+    
+    @Flag(help: "whether to skip the compilation steps and assume that build artifacts from a previous run are still located at the expected places")
+    var awsDeployOnly: Bool = false
+    
+    
+    private(set) lazy var packageRootDir: URL = URL(fileURLWithPath: inputPackageDir).absoluteURL
+    
+    
+    func run() throws {
+        var deploymentProvider = LambdaDeploymentProvider(
+            productName: productName,
+            packageRootDir: URL(fileURLWithPath: inputPackageDir).absoluteURL,
+            awsProfileName: awsProfileName,
+            awsRegion: awsRegion,
+            s3BucketName: s3BucketName,
+            s3BucketPath: s3BucketPath,
+            awsApiGatewayApiId: awsApiGatewayApiId,
+            awsDeployOnly: awsDeployOnly
+        )
+        try deploymentProvider.run()
+    }
+}
+
+
+
+private let dockerfileContents: String = """
+FROM swift:5.3-amazonlinux2
+
+ARG USER_ID
+ARG GROUP_ID
+ARG USERNAME
+
+RUN yum -y install zip sqlite-devel
+
+RUN groupadd --gid $GROUP_ID $USERNAME \
+    && useradd -s /bin/bash --uid $USER_ID --gid $GROUP_ID -m $USERNAME
+
+USER $USERNAME
+"""
+
+
+private let dockerignoreContents: String = """
+.build/
+"""
+
+
+
+private let collectSharedObjectFilesScriptContents: String = """
+#!/bin/bash
+set -eu
+
+executable_path=$1 # path to the built executable
+output_dir=$2      # path of the directory we should copy the object files to
+
+rm -rf "$output_dir"
+mkdir -p "$output_dir"
+# add the target deps based on ldd
+ldd "$executable_path" | grep swift | awk '{print $3}' | xargs cp -L -t "$output_dir"
+"""
```

Comment:
I would suggest using resources for the content of these constants

Reply:
oh wow i had completely missed that apple finally added resource support to SPM. will move to that.